src/plugins/auth/ldap.ts

```diff
@@ -173,16 +183,17 @@ function advancedLdapLoadUsers(users, callbackLoadUser) {
 			filter: `${config.ldap.searchDN.filter}`,
 			attributes: [config.ldap.primaryKey],
 			paged: true,
-		};
+		} as SearchOptions;
 
 		ldapclient.search(base, searchOptions, function (err2, res) {
 			if (err2) {
-				log.error(`LDAP search error: ${err2}`);
+				log.error(`LDAP search error: ${err2?.toString()}`);
 				return true;
 			}
 
 			res.on("searchEntry", function (entry) {
-				const user = entry.attributes[0]._vals[0].toString();
+				// TODO: was _vals.
```

Comment:
todo

Reply:
The official typing is `.vals`. Seems to be a wrapper for the private `_vals`, but I don't use LDAP and can't actually verify.

Reply:
I do think it should be vals, upon looking at the node-ldap code closer. 

<img width="921" alt="image" src="https://user-images.githubusercontent.com/8675906/172755291-00cf125f-246b-40c9-841c-ac63be084456.png">